### PR TITLE
Source Google Search Console: enable `high` test strictness level in SAT

### LINKED ISSUE #19044: Source Google Search Console: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Google Search Console is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by lazebnyi**
https://github.com/airbytehq/airbyte/issues/19044
SAT high strictness

### CODE REVIEW

**Comment by evantahler** [comment]
My review is pending the results of [this slack converstation](https://airbytehq-team.slack.com/archives/C04G4ST1ZB7/p1673999887742139).  These records look like real ad data for Airbyte, which may be sensitive information.

**Review by evantahler — CHANGES_REQUESTED** [review]
Rejecting as expected_records contains sensitive data.

I think the path forward here is, sadly, to bypass the stream's acceptance test.  It seems like too big of a lift at the moment to spin up a test website and run ads pointing at it...

cc @alafanechere

**Review by evantahler — COMMENTED** [review]
```
{"stream": "search_analytics_by_page", "data": {"clicks": 22, "impressions": 760, "ctr": 0.02894736842105263, "position": 2.375, "site_url": "https://airbyte.io/", "search_type": "web", "date": "2021-09-14", "page": "https://airbyte.io/pricing"}, "emitted_at": 1673994671340}
```

Still might be protected business data - @Patsy-Airbyte and @YowanR to decide.

**Comment by evantahler** [comment]
Swapping my review with @YowanR to decide on expected records being secret or not

**Comment by YowanR** [comment]
@lazebnyi I just checked the expected_records. I would consider `"clicks": 22, "impressions": 760,` to be PIIs. Can we skip those? If not, I suggest that we skip enabling high test strictness for this one. I'm checking this with @Patsy-Airbyte and will get back to you on a final decision today.

**Comment by lazebnyi** [comment]
@YowanR From expected records deleted `clicks`, `impressions` and all fields connected with `country`, `website urls` or `type of devices`. Waiting your approve to merge PR.

**Comment by YowanR** [comment]
Looking at `expected_records.jsonl`, the data now seems to be free of both PII and business sensitive information which is great! We should be able to proceed with the merge for now. In parallel, we should flag all these connectors to the Unblocker team so that we can look into creating richer samples without PIIs or Business sensitive info.
I'm curious --  what is the impact of reducing the records in `expected_records.jsonl`? Does it reduce our test coverage?
cc @evantahler

**Review by YowanR — APPROVED** [review]
LGTM!

File: airbyte-integrations/connectors/source-google-search-console/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,37 +1,48 @@
...
+        timeout_seconds: 1800
   full_refresh:
-    - config_path: "secrets/config.json"
-      configured_catalog_path: "integration_tests/catalog.json"
-      timeout_seconds: 1800
+    tests:
+      - config_path: "secrets/config.json"
+        configured_catalog_path: "integration_tests/catalog.json"
+        timeout_seconds: 1800
   incremental:
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-google-search-console/acceptance-test-config.yml:34]: 👍 no bypass reasons!

File: airbyte-integrations/connectors/source-google-search-console/integration_tests/expected_records.jsonl [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{"stream": "sites", "data": {"permissionLevel": "siteOwner"}, "emitted_at": 1674600072721}
+{"stream": "sites", "data": {"permissionLevel": "siteOwner"}, "emitted_at": 1674600073155}
+{"stream": "sitemaps", "data": {"lastSubmitted": "2021-09-10T23:02:22.258Z", "isPending": false, "isSitemapsIndex": false, "type": "sitemap", "warnings": "0", "errors": "0", "contents": [{"type": "web", "submitted": "5690", "indexed": "0"}]}, "emitted_at": 1674600073428}
+{"stream": "sitemaps", "data": {"lastSubmitted": "2021-09-10T23:02:22.258Z", "isPending": false, "isSitemapsIndex": false, "type": "sitemap", "warnings": "0", "errors": "0", "contents": [{"type": "web", "submitted": "5690", "indexed": "0"}]}, "emitted_at": 1674600073798}
+{"stream": "search_analytics_by_date", "data": {"ctr": 0.07940251572327044, "position": 31.88456498951782, "search_type": "web", "date": "2021-09-14"}, "emitted_at": 1674600074364}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-google-search-console/integration_tests/expected_records.jsonl:5]: This data seems vague enough to be OK 👍.  Can you confirm that this is not data about www.airbute.com/www.airbute.io?

**Review by evantahler — APPROVED** [review]

**Pull request merged**